lib/src/backends/record_replay/mutable_recording.dart

```diff
@@ -53,6 +54,20 @@ class MutableRecording implements LiveRecording {
     }
   }
 
+  /// Returns a new file for use with this recording.
+  ///
+  /// The file name will combine the specified [name] with [newUid] to ensure
+  /// that its name is unique among all recording files.
+  ///
+  /// It is up to the caller to create the file - it will not exist in the
+  /// file system when it is returned from this method.
+  File newFile(String name) {
+    String basename = '${new NumberFormat('000').format(newUid())}.$name';
```

Comment:
I'm assuming the  odds of this `NumberFormat` being re-used elsewhere is between low and nil, otherwise consider extracting to a top-level private final.